medical-portal/src/API/RSBC.DMF.MedicalPortal.API/AutoMapper.cs

```diff
@@ -16,9 +25,97 @@ public MappingProfile()
                 .ForMember(dest => dest.FullName, opt => opt.MapFrom(src => src.Case.Person.FullName))
                 .ForMember(dest => dest.Birthday, opt => opt.MapFrom(src => src.Case.Person.Birthday))
                 .ForMember(dest => dest.ComplianceDate, opt => opt.MapFrom(src => src.ComplianceDate));
+
+
+            CreateMap<LegacyDocument, ViewModels.CaseDocument>()
+             .ForMember(dest => dest.SubmittalStatus, opt => opt.MapFrom(src => src.SubmittalStatus))
+             .ForMember(dest => dest.DueDate, opt => opt.MapFrom(src => DueDateConverter(src)))
+             .ForMember(dest => dest.CreatedOn, opt => opt.MapFrom(src => CreateDateConverter(src)))
+             .ForMember(dest => dest.DocumentType, opt => opt.MapFrom(src => src.DocumentType))
+             .ForMember(dest => dest.Description, opt => opt.MapFrom(src => src.Description));
+        }
+
+
+        private Expression<Func<string, string>> NullStringConverter = x => x ?? string.Empty;
+
+        private DateTimeOffset ImportDateConverter(LegacyDocument src)
+        {
+            var importDate = DateTimeOffset.Now;
+            try
+            {
+                if (src.ImportDate != null)
+                {
+                    importDate = src.ImportDate.ToDateTimeOffset();
+                    if (importDate.Offset == TimeSpan.Zero)
+                    {
+                        importDate = TimeZoneInfo.ConvertTimeFromUtc(importDate.DateTime, _pacificZone);
+                    }
+                }
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation(ex, "Error parsing import date");
+                importDate = DateTimeOffset.Now;
+            }
+
+            return importDate;
+        }
+
+
+        //private Expression<Func<string, string>> NullStringConverter = x => x ?? string.Empty;
+
+        private DateTimeOffset DueDateConverter(LegacyDocument src)
```

Comment:
this is not needed, automapper will pickup the Timestamp to DateTimeOffset mapping, no need to manually map this field

Reply:
Removed the date convertor